File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/FriendshipDbEntityMapper.java
package uk.co.victoriajanedavis.chatapp.data.mappers;

import uk.co.victoriajanedavis.chatapp.data.model.db.FriendshipDbModel;
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper;
import uk.co.victoriajanedavis.chatapp.domain.entities.FriendshipEntity;

public class FriendshipDbEntityMapper extends Mapper<FriendshipDbModel, FriendshipEntity> {

    @Override
    public FriendshipEntity mapFrom(FriendshipDbModel from) {
        FriendshipEntity entity = new FriendshipEntity();
        entity.setUuid(from.getUuid());
        entity.setUsername(from.getUsername());
        entity.setEmail(from.getEmail());
        entity.setFriendshipAccepted(from.isFriendshipAccepted());
        entity.setSentFromCurrentUser(from.isSentFromCurrentUser());
        entity.setChatUuid(from.getChatUuid());

        return entity;
    }
}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/mappers/FriendshipDbEntityMapper.kt
package uk.co.victoriajanedavis.chatapp.data.mappers

import uk.co.victoriajanedavis.chatapp.data.model.db.FriendshipDbModel
import uk.co.victoriajanedavis.chatapp.domain.common.Mapper
import uk.co.victoriajanedavis.chatapp.domain.entities.FriendshipEntity

class FriendshipDbEntityMapper : Mapper<FriendshipDbModel, FriendshipEntity>() {

    override fun mapFrom(from: FriendshipDbModel): FriendshipEntity {
        return FriendshipEntity(
            uuid = from.uuid,
            username = from.username,
            email = from.email,
            isFriendshipAccepted = from.isFriendshipAccepted,
            isSentFromCurrentUser = from.isSentFromCurrentUser,
            chatUuid = from.chatUuid,
            loadingState = from.loadingState
        )
    }
}